server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/TransportGetAliasesAction.java

```diff
@@ -63,7 +76,61 @@ protected GetAliasesResponse newResponse() {
     protected void masterOperation(GetAliasesRequest request, ClusterState state, ActionListener<GetAliasesResponse> listener) {
         String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(state, request);
         ImmutableOpenMap<String, List<AliasMetaData>> result = state.metaData().findAliases(request.aliases(), concreteIndices);
-        listener.onResponse(new GetAliasesResponse(result));
+
+        String message = null;
+        RestStatus status = null;
+        if (false == Strings.isAllOrWildcard(request.aliases())) {
+            String[] aliasesNames = Strings.EMPTY_ARRAY;
+
+            if (false == Strings.isAllOrWildcard(request.aliases())) {
+                aliasesNames = request.aliases();
+            }
+
+            final Set<String> aliasNames = new HashSet<>();
+            for (final ObjectObjectCursor<String, List<AliasMetaData>> cursor : result) {
+                for (final AliasMetaData aliasMetaData : cursor.value) {
+                    aliasNames.add(aliasMetaData.alias());
+                }
+            }
+
+            // first remove requested aliases that are exact matches
+            final SortedSet<String> difference = Sets.sortedDifference(Arrays.stream(aliasesNames).collect(Collectors.toSet()), aliasNames);
+
+            // now remove requested aliases that contain wildcards that are simple matches
+            final List<String> matches = new ArrayList<>();
+            outer:
+            for (final String pattern : difference) {
+                if (pattern.contains("*")) {
+                    for (final String aliasName : aliasNames) {
+                        if (Regex.simpleMatch(pattern, aliasName)) {
+                            matches.add(pattern);
+                            continue outer;
+                        }
+                    }
+                }
+            }
+
+            difference.removeAll(matches);
+            if (false == difference.isEmpty()) {
+                status = RestStatus.NOT_FOUND;
+                if (difference.size() == 1) {
+                    message = String.format(Locale.ROOT, "alias [%s] missing", toNamesString(difference.iterator().next()));
+                } else {
+                    message = String.format(Locale.ROOT, "aliases [%s] missing", toNamesString(difference.toArray(new String[0])));
+                }
+            }
+        }
+        listener.onResponse(new GetAliasesResponse(result, status, message));
+    }
+
+    private static String toNamesString(final String... names) {
```

Comment:
nit: would it be possible to use `Strings.collectionToCommaDelimitedString(difference)` instead ?